### PR TITLE
Implement JWT Security

### PR DESCRIPTION
**Pull request by DEz-1996**

### CODE REVIEW

File: src/main/java/online/book/store/controller/AuthenticationController.java [hunk 1]
```diff
@@ -19,6 +22,13 @@
 @Tag(name = "User authentication", description = "Endpoints for authentication users")
 public class AuthenticationController {
     private final UserService userService;
+    private final AuthenticationService authenticationService;
+
+    @PostMapping("/login")
+    @Operation(summary = "Login exist user", description = "Login exist user")
+    public UserLoginResponceDto login(@RequestBody UserLoginRequestDto requestDto) {
```
Comment (andrii-hoienko) [inline src/main/java/online/book/store/controller/AuthenticationController.java]: Looks like you forget to add @Valid annotation

File: src/main/java/online/book/store/dto/UserLoginRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import jakarta.validation.constraints.NotEmpty;
+import jakarta.validation.constraints.Size;
+
+public record UserLoginRequestDto(
+        @NotEmpty
+        @Email
+        @Size(min = 8, max = 20)
+        String email,
+        @NotEmpty
+        @Email
```
Comment (andrii-hoienko) [inline src/main/java/online/book/store/dto/UserLoginRequestDto.java]: Password should be validated as email?

File: src/main/java/online/book/store/security/JwtAuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+                    null,
+                    userDetails.getAuthorities());
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
```
Comment (andrii-hoienko) [inline src/main/java/online/book/store/security/JwtAuthenticationFilter.java]: Please make "Bearer " constant, then you will be able to get rid of BEARER_START_INDEX_TOKEN constant, becasue 7 is the length of "Bearer " string

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]

**Review by vovasalyha — APPROVED** [review]
GJ, but pay attention to commit messages. `Some fixes with mentor's advice` is not very informative.

**Pull request merged**